rust_src/src/fonts.rs

```diff
@@ -144,31 +144,20 @@ impl From<LispFontObjectRef> for LispObject {
 
 impl From<LispObject> for LispFontObjectRef {
     fn from(o: LispObject) -> Self {
-        o.as_font_object_or_error()
+        Option::<LispFontObjectRef>::from(o).unwrap_or_else(|| wrong_type!(Qfont_object, o))
```

Comment:
If that felt ugly you could write:

    match o.into() {
        Some(font) => font,
        None => wrong_type!(Qfont_object, o),
    }

Reply:
Ah yes, I missed that. That's a lot less ugly.